sycl/test/matrix/matrix-bfloat16-test-coord-basic.cpp

```diff
@@ -0,0 +1,417 @@
+
+using namespace sycl::ext::oneapi::experimental::matrix;
+using bfloat16 = sycl::ext::oneapi::experimental::bfloat16;
+
+#define SG_SZ 16
+
+#define TM 8
+#define TN SG_SZ
+#define TK 16
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+static constexpr size_t MATRIX_M = 2 * TM;
+static constexpr size_t MATRIX_N = 2 * TN;
+static constexpr size_t MATRIX_K = 2 * TK;
+bfloat16 A_ref[MATRIX_M][MATRIX_K];
+bfloat16 B_ref[MATRIX_K][MATRIX_N];
+float C_ref[MATRIX_M][MATRIX_N];
+
+bfloat16 *A;
+bfloat16 *B;
+float *C;
+
+// float *res_local_rowA;
+// float *res_local_colB;
+// float *res_local_rowC;
+
+float *res_local_row_origA;
+float *res_local_col_origB;
+float *res_local_row_origC;
+
+void matrix_coord(sycl::queue &q, float *resA, float *resB, float *resC) {
+
+  size_t NDRangeM = MATRIX_M / TM;
+  size_t NDRangeN = MATRIX_N / TN;
+
+  auto pA =
+      sycl::multi_ptr<bfloat16, sycl::access::address_space::global_space>(A);
+  auto pB =
+      sycl::multi_ptr<bfloat16, sycl::access::address_space::global_space>(B);
+  auto pC =
+      sycl::multi_ptr<float, sycl::access::address_space::global_space>(C);
+
+  q.submit([&](sycl::handler &cgh) {
+     cgh.parallel_for<class imatrix>(
+         sycl::nd_range<2>({NDRangeM, NDRangeN * SG_SZ}, {1, 1 * SG_SZ}), [=
+     ](sycl::nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]]
+
+         {
+           size_t M = MATRIX_M;
+           size_t N = MATRIX_N;
+           size_t K = MATRIX_K;
+
+           sycl::ext::oneapi::sub_group sg = spmd_item.get_sub_group();
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           joint_matrix<bfloat16, TM, TK, use::a> sub_a(sg);
+           joint_matrix<bfloat16, TK, TN, use::b> sub_b(sg);
+           joint_matrix<float, TM, TN, use::accumulator> sub_c(sg);
+
+           joint_matrix_load(sg, sub_c,
+                             pC + (sg_startx * TM) * N + sg_starty / SG_SZ * TN,
+                             N, layout::row_major);
+           auto tCData = sub_c.get_wi_data();
+           for (int i = 0; i < tCData.length(); ++i) {
+             auto [row, col] = tCData[i].get_coord();
+             resC[row] += tCData[i];
+           }
+
+           for (int k = 0; k < K / TK; k += 1) { //
+             joint_matrix_load(sg, sub_a, pA + (sg_startx * TM) * K + k * TK, K,
+                               layout::row_major);
+             auto tAData = sub_a.get_wi_data();
+             for (int i = 0; i < tAData.length(); ++i) {
+               auto [row, col] = tAData[i].get_coord();
+               resA[row] += tAData[i];
+             }
```

Comment:
// SG size = 64, sub_a[8][16], multiple WI share a row
WI 0 --> 2 elements of row 0 --> resA[0]
WI 1 --> 3 elements of row 0 --> resA[0]
resA should be private variable --> length size
partial_sum[row] = reduction among the WI (reduce_over_group), 
every WI will have same value of partial_sum[row]
copy partial_sum into the global variable